contracts/governance/FeeSharingProxy.sol

```diff
@@ -338,6 +365,33 @@ contract FeeSharingProxy is SafeMath96, IFeeSharingProxy {
 		}
 		emit CheckpointAdded(msg.sender, _token, _numTokens);
 	}
+
+	/**
+	 * @notice Initialise previous accumulated rewards of a user.
+	 *
+	 * The Sovryn protocol collects fees on every trade/swap and loan.
+	 * These fees will be distributed to only SOV stakers instead of
+	 * both vesters and stakers to increase rewards of stakers. To keep
+	 * it fair, previous rewards are initialised and will be available
+	 * for claim.
+	 *
+	 * @param account Address of the vesting user
+	 * @param amount amount of tokens accumulated as reward
+	 * @param vestingRewardsUptoBlock The block number
+	 * */
+	function initialiseVestingRewards(
+		address account,
+		uint96 amount,
+		uint32 vestingRewardsUptoBlock
+	) external onlyOwner {
+		require(account != address(0), "Staking::initialiseVestingRewards: invalid account");
+		require(amount != 0, "Staking::initialiseVestingRewards: zero amount");
+		require(vestingRewardsUptoBlock < block.number, "Staking::initialiseVestingRewards: invalid block number");
+		VestingRewards memory vestingRewards;
+		vestingRewards.rewardsUptoBlock = vestingRewardsUptoBlock;
+		vestingRewards.amount = amount;
+		previousVestingRewards[account] = vestingRewards;
+	}
```

Comment:
this is dirty.
it means that
a) the owner can set overwrite all existing rewards for  vesting contracts
b) we need to call this function after the update for all vesting contracts
c) there are unused parameters on the vesting rewards
d) instead of having this dirty initialization, we could just store the block number until which vesting contracts get rewarded and only iterate from the last checkpoint until this number in case the user address is a vesting contract